src/auth/login/LoginMsal.tsx

```diff
@@ -0,0 +1,102 @@
+import { useMsal, MsalProvider, useIsAuthenticated } from "@azure/msal-react";
+import { PublicClientApplication } from "@azure/msal-browser";
+import { loginRequest } from "../models/authConfig";
+import { msalConfig } from "../models/authConfig";
+import type { LoginInfo } from "../models/LoginInfo";
+import { useEffect, useState } from "react";
+import LoginCard from "./components/LoginCard";
+import useTokenExchange from "./hooks/useTokenExchange";
+import { useNeedsManualApproval } from "./hooks/useNeedsManualApproval";
+
+const MsalLoginInternal = ({ info }: { info: LoginInfo }) => {
+  const { instance, accounts } = useMsal();
+  const [accessToken, setAccessToken] = useState<string>("");
+  const isAuthenticated = useIsAuthenticated();
+
+  const { needsManualApproval, onApprovalReceived } =
+    useNeedsManualApproval(info);
```

Comment:
@bakerpdgit this new logic should ensure that we do not ask for consent on perse.pythonsponge.com, or on any other domain where the book's domain matches the frontend's domain.